sdk/storage/sample/datalake_getting_started.cpp

```diff
@@ -3,11 +3,80 @@
 
 #include "datalake/datalake.hpp"
 #include "samples_common.hpp"
+#include <iostream>
 
 SAMPLE(DataLakeGettingStarted, DataLakeGettingStarted)
 void DataLakeGettingStarted()
 {
+  // Please note that you can always reference test cases for advanced usages.
+
   using namespace Azure::Storage::Files::DataLake;
-  auto client = ServiceClient::CreateFromConnectionString(GetConnectionString());
-  auto response = client.ListFileSystems();
+
+  std::string fileSystemName = "sample-file-system";
+  std::string directoryName = "sample-directory";
+  std::string fileName = "sample-file";
+
+  // Initializing a ServiceClient that can then initialize the FileSystemClient or list file
+  // systems.
+  auto serviceClient = ServiceClient::CreateFromConnectionString(GetConnectionString());
+
+  // List all file systems.
+  std::string continuation;
+  std::vector<FileSystem> fileSystems;
+  do
+  {
+    auto response = serviceClient.ListFileSystems();
+    if (response->Continuation.HasValue())
+    {
+      continuation = response->Continuation.GetValue();
+    }
+    fileSystems.insert(
+        fileSystems.end(), response->Filesystems.begin(), response->Filesystems.end());
+  } while (!continuation.empty());
+
+  // Create file systems.
+  auto fileSystemClient = serviceClient.GetFileSystemClient(fileSystemName);
+  auto iter = std::find_if(
+      fileSystems.begin(), fileSystems.end(), [&fileSystemName](const FileSystem& fileSystem) {
+        return fileSystem.Name == fileSystemName;
+      });
+  if (iter == fileSystems.end())
+  {
+    fileSystemClient.Create();
+  }
+
+  // Create a directory.
+  auto directoryClient = fileSystemClient.GetDirectoryClient(directoryName);
+  directoryClient.Create();
+
+  // Creates a file under the directory.
+  auto fileClient = fileSystemClient.GetFileClient(directoryName + "/" + fileName);
+  fileClient.Create();
+
+  // Append/flush/read data from the client.
+  // Append data
+  size_t bufferSize = 64;
+  std::vector<uint8_t> buffer(bufferSize);
+  for (size_t i = 0; i < bufferSize; ++i)
+  {
+    buffer[i] = static_cast<uint8_t>(i >> 3);
+  }
+  auto bufferStream = std::make_unique<Azure::Core::Http::MemoryBodyStream>(
```

Comment:
Here you're trying to give users an impression that bodyStream must be allocated on heap, or managed by smart pointer, which is not the case. We've decided that input bodyStream takes a raw pointer, so that our users can have more flexible memory management methods, which means they can choose to allocate it on stack or on heap, to manage it via raw pointer or leave it to smart pointer.

Sample file is usually the first place our new users are going to look at. So it's very important for us to convey accurate information. It's not like unittest code, which is fine as long as it works.

Reply:
I don't think it should be covered by Storage's sample, if we need to cover every bit of usage of Azure Core, the effort will be huge for us.
Anyway, I think it is not a bad idea to provide more information in the sample, so I changed how `MemoryBodyStream` can be initialized and showed two approaches.

Reply:
I didn't mean we should cover everything of storage, not at all. We can only convey very limited information from sample, so we should make it concise, accurate and less misunderstanding.

Reply:
> so we should make it concise, accurate and less misleading.

Well, I think it is at least equally important for our customers to know the buffer can be passed in by either a raw pointer or reference, so I made the sample to cover both. I still don't think the previous implementation is misleading, as we can always use a dereference operator or a reference operator to transform one into another.

Reply:
i'm sure we're not talking about the same thing.

Reply:
I'm concerned about `MemoryBodyStream` being managed by `std::unique_ptr`, not which constructor of `MemoryBodyStream` you're calling.

Reply:
I see, then I actually agree with you. Changed to stack-allocated.